src/backend/cgelem.c

```diff
@@ -3555,8 +3555,8 @@ STATIC elem * eleq(elem *e, goal_t goal)
             {
                 e->E2 = e2->E1;
                 eb = el_bin(OPeq,ty,eb,e2->E2);
-                e2->E1 = eb;
-                e2->E2 = e;
+                e2->E1 = e;
+                e2->E2 = eb;
```

Comment:
So the two leaves are executed in the same order as in the OPpair expression node.